pkg/providers/tinkerbell/config/template-cp.yaml

```diff
@@ -178,6 +178,12 @@ spec:
   template:
     {{- if and .etcdTemplateOverride (ne .format "")}}
     spec:
+      hardwareAffinity:
+        required:
+        - labelSelector:
+            matchLabels: {{ range $key, $value := .etcdHardwareSelector}}
+              {{ $key }}: {{ $value}}
+            {{- end }}
```

Comment:
Have you checked what happens here if the `etcdHardwareSelector` is empty?

Reply:
I have added a basic validation where HardwareSelector is mandatory for each machine config for Tinkerbell.